### PR TITLE
CORTX-29526: fix memleaks at data-path

### PR DESCRIPTION
**Pull request by andriytk**
# Problem Statement
There are memory leaks at `dix_cas_rops_send()` and `m0__obj_namei_send()` and `sync_pending_stx_update()` reported by valgrind when doing S3 object put/get/del operations with rgw:

```
+==15225== 144 bytes in 3 blocks are definitely lost 
+==15225==    at 0x4C360A5: malloc (vg_replace_malloc.c:380)
+==15225==    by 0xBE4FE3C: m0_alloc (memory.c:133)
+==15225==    by 0xBE98059: sync_pending_stx_update (sync.c:772)
+==15225==    by 0xBE987E4: sync_record_update (sync.c:807)
+==15225==    by 0xBE83435: ioreq_cc_bottom_half (io_req_fop.c:500)
+==15225==    by 0xBEF2FFE: m0_sm_asts_run (sm.c:175)
+==15225==    by 0xBEF2FFE: m0_sm_asts_run (sm.c:150)
+==15225==    by 0xBE1BAC1: loc_handler_thread (fom.c:925)
+==15225==    by 0xBE5174D: m0_thread_trampoline (thread.c:117)
+==15225==    by 0xBE5CDC0: uthread_trampoline (uthread.c:98)
+==15225==    by 0x8AE0179: start_thread (in /usr/lib64/libpthread-2.28.so)
+==15225==    by 0xA418DF2: clone (in /usr/lib64/libc-2.28.so)
+==15225== 
+==15225== 144 bytes in 3 blocks are definitely lost 
+==15225==    at 0x4C360A5: malloc (vg_replace_malloc.c:380)
+==15225==    by 0xBE4FE3C: m0_alloc (memory.c:133)
+==15225==    by 0xBE98059: sync_pending_stx_update (sync.c:772)
+==15225==    by 0xBE987E4: sync_record_update (sync.c:807)
+==15225==    by 0xBE83862: io_bottom_half (io_req_fop.c:333)
+==15225==    by 0xBEF2FFE: m0_sm_asts_run (sm.c:175)
+==15225==    by 0xBEF2FFE: m0_sm_asts_run (sm.c:150)
+==15225==    by 0xBE1BAC1: loc_handler_thread (fom.c:925)
+==15225==    by 0xBE5174D: m0_thread_trampoline (thread.c:117)
+==15225==    by 0xBE5CDC0: uthread_trampoline (uthread.c:98)
+==15225==    by 0x8AE0179: start_thread (in /usr/lib64/libpthread-2.28.so)
+==15225==    by 0xA418DF2: clone (in /usr/lib64/libc-2.28.so)
+==15225== 
+==15225== 720 bytes in 15 blocks are definitely lost 
+==15225==    at 0x4C360A5: malloc (vg_replace_malloc.c:380)
+==15225==    by 0xBE4FE3C: m0_alloc (memory.c:133)
+==15225==    by 0xBE98059: sync_pending_stx_update (sync.c:772)
+==15225==    by 0xBE987E4: sync_record_update (sync.c:807)
+==15225==    by 0xBDED25C: dix_cas_rop_clink_cb (req.c:1714)
+==15225==    by 0xBE4C22E: clink_signal (chan.c:135)
+==15225==    by 0xBE4C2A9: chan_signal_nr (chan.c:154)
+==15225==    by 0xBEF469F: state_set (sm.c:462)
+==15225==    by 0xBDAADEA: cas_req_state_set (client.c:246)
+==15225==    by 0xBDAC234: cas_req_replied_ast (client.c:1160)
+==15225==    by 0xBEF2FFE: m0_sm_asts_run (sm.c:175)
+==15225==    by 0xBEF2FFE: m0_sm_asts_run (sm.c:150)
+==15225==    by 0xBE1BAC1: loc_handler_thread (fom.c:925)
+==15225== 
+==15225== 1,299 (648 direct, 651 indirect) bytes in 3 blocks are definitely lost 
+==15225==    at 0x4C360A5: malloc (vg_replace_malloc.c:380)
+==15225==    by 0xBE4FE3C: m0_alloc (memory.c:133)
+==15225==    by 0xBE8C2AB: cob_req_alloc.isra.9 (cob.c:310)
+==15225==    by 0xBE8E52F: m0__obj_namei_send (cob.c:1825)
+==15225==    by 0xBE8F491: obj_namei_cb_launch (obj.c:263)
+==15225==    by 0xBE8F491: obj_namei_cb_launch (obj.c:219)
+==15225==    by 0xBE73142: m0_op_launch_one (client.c:719)
+==15225==    by 0xBE732BB: m0_op_launch (client.c:733)
+==15225==    by 0x5AC8F92: rgw::sal::MotrObject::open_mobj(DoutPrefixProvider const*) (rgw_sal_motr.cc:1745)
+==15225==    by 0x5ADB64A: rgw::sal::MotrObject::MotrReadOp::prepare(optional_yield, DoutPrefixProvider const*) (rgw_sal_motr.cc:1446)
+==15225==    by 0x57C5967: RGWGetObj::execute(optional_yield) (rgw_op.cc:2151)
+==15225==    by 0x53D7AB5: rgw_process_authenticated(RGWHandler_REST*, RGWOp*&, RGWRequest*, req_state*, optional_yield, rgw::sal::Store*, bool) (rgw_process.cc:254)
+==15225==    by 0x53DA564: process_request(rgw::sal::Store*, RGWREST*, RGWRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rgw::auth::StrategyRegistry const&, RGWRestfulIO*, OpsLogSink*, optional_yield, rgw::dmclock::Scheduler*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::chrono::duration<unsigned long, std::ratio<1l, 1000000000l> >*, std::shared_ptr<RateLimiter>, int*) (rgw_process.cc:394)
+==15225== 
+==15225== 1,299 (648 direct, 651 indirect) bytes in 3 blocks are definitely lost 
+==15225==    at 0x4C360A5: malloc (vg_replace_malloc.c:380)
+==15225==    by 0xBE4FE3C: m0_alloc (memory.c:133)
+==15225==    by 0xBE8C2AB: cob_req_alloc.isra.9 (cob.c:310)
+==15225==    by 0xBE8E52F: m0__obj_namei_send (cob.c:1825)
+==15225==    by 0xBE8F491: obj_namei_cb_launch (obj.c:263)
+==15225==    by 0xBE8F491: obj_namei_cb_launch (obj.c:219)
+==15225==    by 0xBE73142: m0_op_launch_one (client.c:719)
+==15225==    by 0xBE732BB: m0_op_launch (client.c:733)
+==15225==    by 0x5AC8F92: rgw::sal::MotrObject::open_mobj(DoutPrefixProvider const*) (rgw_sal_motr.cc:1745)
+==15225==    by 0x5AC97D2: rgw::sal::MotrObject::delete_mobj(DoutPrefixProvider const*) (rgw_sal_motr.cc:1775)
+==15225==    by 0x5ACACF5: rgw::sal::MotrObject::MotrDeleteOp::delete_obj(DoutPrefixProvider const*, optional_yield) (rgw_sal_motr.cc:1540)
+==15225==    by 0x57EC2BF: RGWDeleteObj::execute(optional_yield) (rgw_op.cc:5032)
+==15225==    by 0x53D7AB5: rgw_process_authenticated(RGWHandler_REST*, RGWOp*&, RGWRequest*, req_state*, optional_yield, rgw::sal::Store*, bool) (rgw_process.cc:254)
+==15225== 
+==15225== 1,542 (648 direct, 894 indirect) bytes in 3 blocks are definitely lost 
+==15225==    at 0x4C360A5: malloc (vg_replace_malloc.c:380)
+==15225==    by 0xBE4FE3C: m0_alloc (memory.c:133)
+==15225==    by 0xBE8C2AB: cob_req_alloc.isra.9 (cob.c:310)
+==15225==    by 0xBE8E52F: m0__obj_namei_send (cob.c:1825)
+==15225==    by 0xBE8F491: obj_namei_cb_launch (obj.c:263)
+==15225==    by 0xBE8F491: obj_namei_cb_launch (obj.c:219)
+==15225==    by 0xBE73142: m0_op_launch_one (client.c:719)
+==15225==    by 0xBE732BB: m0_op_launch (client.c:733)
+==15225==    by 0x5AB7E32: rgw::sal::MotrObject::create_mobj(DoutPrefixProvider const*, unsigned long) (rgw_sal_motr.cc:1688)
+==15225==    by 0x5ACB53A: rgw::sal::MotrAtomicWriter::write() (rgw_sal_motr.cc:2296)
+==15225==    by 0x5ACBBE3: rgw::sal::MotrAtomicWriter::process(ceph::buffer::v15_2_0::list&&, unsigned long) (rgw_sal_motr.cc:2364)
+==15225==    by 0x57D343D: RGWPutObj::execute(optional_yield) (rgw_op.cc:4089)
+==15225==    by 0x53D7AB5: rgw_process_authenticated(RGWHandler_REST*, RGWOp*&, RGWRequest*, req_state*, optional_yield, rgw::sal::Store*, bool) (rgw_process.cc:254)
+==15225== 
+==15225== 2,208 bytes in 46 blocks are definitely lost 
+==15225==    at 0x4C360A5: malloc (vg_replace_malloc.c:380)
+==15225==    by 0xBE4FE3C: m0_alloc (memory.c:133)
+==15225==    by 0xBDEB356: m0_dix_imask_copy (imask.c:178)
+==15225==    by 0xBDEE996: dix_cas_rops_send (req.c:1774)
+==15225==    by 0xBDEF843: dix__rop (req.c:1386)
+==15225==    by 0xBDEFE9E: dix_rop (req.c:1402)
+==15225==    by 0xBDEFE9E: dix_discovery_completed (req.c:1121)
+==15225==    by 0xBDF08DF: dix_layout_find_ast_cb (req.c:364)
+==15225==    by 0xBEF2FFE: m0_sm_asts_run (sm.c:175)
+==15225==    by 0xBEF2FFE: m0_sm_asts_run (sm.c:150)
+==15225==    by 0xBE1BAC1: loc_handler_thread (fom.c:925)
+==15225==    by 0xBE5174D: m0_thread_trampoline (thread.c:117)
+==15225==    by 0xBE5CDC0: uthread_trampoline (uthread.c:98)
+==15225==    by 0x8AE0179: start_thread (in /usr/lib64/libpthread-2.28.so)
+==15225== 
+==15225== 2,544 bytes in 53 blocks are definitely lost 
+==15225==    at 0x4C360A5: malloc (vg_replace_malloc.c:380)
+==15225==    by 0xBE4FE3C: m0_alloc (memory.c:133)
+==15225==    by 0xBDEB356: m0_dix_imask_copy (imask.c:178)
+==15225==    by 0xBDEE996: dix_cas_rops_send (req.c:1774)
+==15225==    by 0xBDEF843: dix__rop (req.c:1386)
+==15225==    by 0xBDEFE9E: dix_rop (req.c:1402)
+==15225==    by 0xBDEFE9E: dix_discovery_completed (req.c:1121)
+==15225==    by 0xBDF05F7: dix_discovery_ast (req.c:1140)
+==15225==    by 0xBEF2FFE: m0_sm_asts_run (sm.c:175)
+==15225==    by 0xBEF2FFE: m0_sm_asts_run (sm.c:150)
+==15225==    by 0xBE1BAC1: loc_handler_thread (fom.c:925)
+==15225==    by 0xBE5174D: m0_thread_trampoline (thread.c:117)
+==15225==    by 0xBE5CDC0: uthread_trampoline (uthread.c:98)
+==15225==    by 0x8AE0179: start_thread (in /usr/lib64/libpthread-2.28.so)
```

# Solution
Fix the leaks by:
 - clean-up local `m0_cas_id` structure at `dix_cas_rops_send()`;
 - clean-up `struct cob_req` at `m0__obj_namei_send()` in case when metadata lookup is skipped and cob_req is not sent anywhere;
 - clean-up op_pending_tx at m0_op_fini().

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by cortx-admin** [comment]
<h1>Jenkins CI Result : <a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132/>Motr#1132</a></h1><h2>Motr Test Summary</h2><table><tr><th>Test Result</th><th>Count</th><th>Info</tr><tr><td>:x:Failed</td><td>2</td><td><details><summary>:file_folder:</summary><p><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/01motr-single-node/52motr-singlenode-sanity target=_blank>01motr-single-node/52motr-singlenode-sanity</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/01motr-single-node/00userspace-tests target=_blank>01motr-single-node/00userspace-tests</a></p></details></td></tr><tr><td>:checkered_flag:Skipped</td><td>32</td><td><details><summary>:file_folder:</summary><p><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/01motr-single-node/28sys-kvs target=_blank>01motr-single-node/28sys-kvs</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/01motr-single-node/35m0singlenode target=_blank>01motr-single-node/35m0singlenode</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/01motr-single-node/04initscripts target=_blank>01motr-single-node/04initscripts</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/01motr-single-node/37protocol target=_blank>01motr-single-node/37protocol</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/02motr-single-node/51kem target=_blank>02motr-single-node/51kem</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/02motr-single-node/20rpc-session-cancel target=_blank>02motr-single-node/20rpc-session-cancel</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/02motr-single-node/10pver-assign target=_blank>02motr-single-node/10pver-assign</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/02motr-single-node/21fsync-single-node target=_blank>02motr-single-node/21fsync-single-node</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/02motr-single-node/13dgmode-io target=_blank>02motr-single-node/13dgmode-io</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/02motr-single-node/14poolmach target=_blank>02motr-single-node/14poolmach</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/02motr-single-node/11m0t1fs target=_blank>02motr-single-node/11m0t1fs</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/02motr-single-node/26motr-user-kernel-tests target=_blank>02motr-single-node/26motr-user-kernel-tests</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/02motr-single-node/08spiel target=_blank>02motr-single-node/08spiel</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/03motr-single-node/06conf target=_blank>03motr-single-node/06conf</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/03motr-single-node/36spare-reservation target=_blank>03motr-single-node/36spare-reservation</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/04motr-single-node/34sns-repair-1n-1f target=_blank>04motr-single-node/34sns-repair-1n-1f</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/04motr-single-node/08spiel-sns-repair-quiesce target=_blank>04motr-single-node/08spiel-sns-repair-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/04motr-single-node/28sys-kvs-kernel target=_blank>04motr-single-node/28sys-kvs-kernel</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/04motr-single-node/11m0t1fs-rconfc-fail target=_blank>04motr-single-node/11m0t1fs-rconfc-fail</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/04motr-single-node/08spiel-sns-repair target=_blank>04motr-single-node/08spiel-sns-repair</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/04motr-single-node/19sns-repair-abort target=_blank>04motr-single-node/19sns-repair-abort</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/04motr-single-node/22sns-repair-ios-fail target=_blank>04motr-single-node/22sns-repair-ios-fail</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/18sns-repair-quiesce target=_blank>05motr-single-node/18sns-repair-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/12fwait target=_blank>05motr-single-node/12fwait</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/16sns-repair-multi target=_blank>05motr-single-node/16sns-repair-multi</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/07mount-fail target=_blank>05motr-single-node/07mount-fail</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/15sns-repair-single target=_blank>05motr-single-node/15sns-repair-single</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/23sns-abort-quiesce target=_blank>05motr-single-node/23sns-abort-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/17sns-repair-concurrent-io target=_blank>05motr-single-node/17sns-repair-concurrent-io</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/07mount target=_blank>05motr-single-node/07mount</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/07mount-multiple target=_blank>05motr-single-node/07mount-multiple</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/12fsync target=_blank>05motr-single-node/12fsync</a></p></details></td></tr><tr><td>:heavy_check_mark:Passed</td><td>39</td><td><details><summary>:file_folder:</summary><p><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/01motr-single-node/43m0crate target=_blank>01motr-single-node/43m0crate</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/01motr-single-node/05confgen target=_blank>01motr-single-node/05confgen</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/01motr-single-node/06hagen target=_blank>01motr-single-node/06hagen</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/01motr-single-node/01net target=_blank>01motr-single-node/01net</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/01motr-single-node/01kernel-tests target=_blank>01motr-single-node/01kernel-tests</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/01motr-single-node/03console target=_blank>01motr-single-node/03console</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/01motr-single-node/02rpcping target=_blank>01motr-single-node/02rpcping</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/02motr-single-node/07m0d-fatal target=_blank>02motr-single-node/07m0d-fatal</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/02motr-single-node/67fdmi-plugin-multi-filters target=_blank>02motr-single-node/67fdmi-plugin-multi-filters</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/02motr-single-node/53clusterusage-alert target=_blank>02motr-single-node/53clusterusage-alert</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/02motr-single-node/41motr-conf-update target=_blank>02motr-single-node/41motr-conf-update</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/03motr-single-node/61sns-repair-motr-1n-1f target=_blank>03motr-single-node/61sns-repair-motr-1n-1f</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/03motr-single-node/08spiel-multi-confd target=_blank>03motr-single-node/08spiel-multi-confd</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/03motr-single-node/69sns-repair-motr-quiesce target=_blank>03motr-single-node/69sns-repair-motr-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/03motr-single-node/62sns-repair-motr-mf target=_blank>03motr-single-node/62sns-repair-motr-mf</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/03motr-single-node/70sns-failure-after-repair-quiesce target=_blank>03motr-single-node/70sns-failure-after-repair-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/03motr-single-node/63sns-repair-motr-1k-1f target=_blank>03motr-single-node/63sns-repair-motr-1k-1f</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/03motr-single-node/60sns-repair-motr-1f target=_blank>03motr-single-node/60sns-repair-motr-1f</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/03motr-single-node/66sns-repair-motr-abort-quiesce target=_blank>03motr-single-node/66sns-repair-motr-abort-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/03motr-single-node/24motr-dix-repair-lookup-insert-spiel target=_blank>03motr-single-node/24motr-dix-repair-lookup-insert-spiel</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/03motr-single-node/68sns-repair-motr-shutdown target=_blank>03motr-single-node/68sns-repair-motr-shutdown</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/03motr-single-node/64sns-repair-motr-ios-fail target=_blank>03motr-single-node/64sns-repair-motr-ios-fail</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/03motr-single-node/24motr-dix-repair-lookup-insert-m0repair target=_blank>03motr-single-node/24motr-dix-repair-lookup-insert-m0repair</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/03motr-single-node/04sss target=_blank>03motr-single-node/04sss</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/03motr-single-node/65sns-repair-motr-abort target=_blank>03motr-single-node/65sns-repair-motr-abort</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/04motr-single-node/48motr-raid0-io target=_blank>04motr-single-node/48motr-raid0-io</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/04motr-single-node/49motr-rpc-cancel target=_blank>04motr-single-node/49motr-rpc-cancel</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/04motr-single-node/25m0kv target=_blank>04motr-single-node/25m0kv</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/04motr-single-node/44motr-rm-lock-cc-io target=_blank>04motr-single-node/44motr-rm-lock-cc-io</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/04motr-single-node/45motr-rmw target=_blank>04motr-single-node/45motr-rmw</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/23dix-repair-m0repair target=_blank>05motr-single-node/23dix-repair-m0repair</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/43motr-sync-replication target=_blank>05motr-single-node/43motr-sync-replication</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/42motr-utils target=_blank>05motr-single-node/42motr-utils</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/45motr-sns-repair-N-1 target=_blank>05motr-single-node/45motr-sns-repair-N-1</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/40motr-dgmode target=_blank>05motr-single-node/40motr-dgmode</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/23dix-repair-quiesce-m0repair target=_blank>05motr-single-node/23dix-repair-quiesce-m0repair</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/23spiel-dix-repair-quiesce target=_blank>05motr-single-node/23spiel-dix-repair-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/44motr-sns-repair target=_blank>05motr-single-node/44motr-sns-repair</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit/05motr-single-node/23spiel-dix-repair target=_blank>05motr-single-node/23spiel-dix-repair</a></p></details></td></tr><tr><td>Total</td><td>73</td><td><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1132//testReport/junit>:link:</a></td></tr></table><h2> CppCheck Summary</h2><p> &emsp;&emsp;&emsp;Cppcheck: No new warnings found :thumbsup:</p>

File: motr/client.c [hunk 1]
```diff
@@ -428,7 +428,7 @@ void m0_obj_init(struct m0_obj *obj,
 
 #ifdef OSYNC
 	m0_mutex_init(&obj->ob_pending_tx_lock);
-	ospti_tlist_init(&obj->ob_pending_tx);
+	spti_tlist_init(&obj->ob_pending_tx);
```
Comment (huanghua78) [inline motr/client.c:431]: This was a typo bug.
Why is it not found before? Because this condition is never satisfied?

**Review by huanghua78 — APPROVED** [review]

**Review by siningwuseagate — APPROVED** [review]

File: motr/client.c (hunk 1 above)
Comment (andriytk) [reply motr/client.c:431]: I suppose so, yes. Thanks @huanghua78 for review!

File: motr/client.c [hunk 2]
```diff
@@ -862,6 +863,11 @@ void m0_op_fini(struct m0_op *op)
 		oc->oc_cb_fini(oc);
 	m0_mutex_fini(&op->op_priv_lock);
 
+	m0_tl_teardown(spti, &op->op_pending_tx, iter)
+		m0_free0(&iter);
+	spti_tlist_fini(&op->op_pending_tx);
+	m0_mutex_fini(&op->op_pending_tx_lock);
+
```
Comment (nikitadanilov) [inline motr/client.c:870]: This change uncovers a bug in m0_op_get() and breaks motr/ut/client.c:ut_test_m0_op_fini(). See https://github.com/Seagate/cortx-motr/pull/1599.

File: dix/req.c [hunk 3]
```diff
@@ -1799,6 +1799,7 @@ static int dix_cas_rops_send(struct m0_dix_req *req)
 		default:
 			M0_IMPOSSIBLE("Unknown req type %u", req->dr_type);
 		}
+		m0_cas_id_fini(&cctg_id);
```
Comment (nikitadanilov) [inline dix/req.c:1802]: This frees used data: m0_cas_get()->m0_cas__get()->cas_req_prep()->cas_records_op_prepare() copies cas_id:

op->cg_id = *index;

This means that extent op->cg_id.ci.layout.dl_desc.ld_imask.im_range is shared between op->cg_id and cctg_id. m0_cas_id_fini() frees the extent, so garbage is written in fop and sent to the server.

Please revert.

Comment (andriytk) [reply dix/req.c:1802]: Opened https://github.com/Seagate/cortx-motr/issues/1601 for this. Thanks!

**Pull request merged**